### DISCUSSION TITLE
useMutation callback  can't catch the error

### DISCUSSION THREAD

**Discussion by onlyhasbi** [TanStack/query#4420] [Q&A] [answered]
Below is the piece of my code that shows an error on the console. and every time this function is triggered, it will show an error in the console from mutate.ts:245

`export function useAdd(value) {
  const queryClient = useQueryClient();
  queryClient.setMutationDefaults([add], {
    mutationFn: async (value) =>
      await http.post(endpoint, value).then(({ data }) => data),
  });

  return useMutation([add], {
    onError: (error, _, __) => {
      toast.error(error.response.data);
    },
    onSuccess: () => {
      toast.success("success");
    },
  });
}`

I call the code above on another file
`const { mutate: add } = useAdd();`

but when I add extra catch
`await http.post(endpoint, value).then(({ data }) => data).catch(error => error)
`
error in console gone, but always showing success result

how it can be ?

**Comment by TkDodo** [comment, marked as answer]
in development mode, errors are logged to the console.

**Reply by onlyhasbi** [reply]
Thx, @TkDodo
